viewing_party/party.py

```diff
@@ -1,23 +1,239 @@
+    
+    most_watched_genre = ""
+    most_watched_genre_count = 0
+    for genre, count in genre_frequency.items():
+        if count > most_watched_genre_count:
+            most_watched_genre = genre
+            most_watched_genre_count = count
+    
+    return most_watched_genre
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    """
+    Return a list of movies that appear in the user's watched list
+    but are not present in any friend's watched list.
+    Each movie is represented as a dictionary with title, genre, and rating.
+    """
+    unique_movies = []
+    user_watched = user_data["watched"]
+    friends = user_data["friends"]
+
+    friends_title = set()
+    for watched_dict in friends:
+        for movie in watched_dict["watched"]:
+            title = movie["title"]
+            if title:
+                friends_title.add(title)
+
+    watched = set()
+    for movie in user_watched:
+        title = movie["title"]
+        if title and title not in friends_title and title not in watched:
+            unique_movies.append(movie)
+            watched.add(title)
+    return unique_movies  
+
+def get_friends_unique_watched(user_data):
+    """Synthesizes all the movies a group of friends have watched into a single list of movies (dictionaries) that the user has not watched and at least one of the user's friends has watched.
+
+    Args:
+        user_data (dictionary):
+            "watched": List of dictionaries that represent movies the user has watched.
+                Movie dictionaries has the following key-value pairs:
+                    title (str)
+                    genre (str)
+                    rating (float)
+            "friends": A list of dictionaries. Each dictionary is a friend.
+                Friend dictionaries have the following key-value pairs:
+                    "watched": List of dictionaries that represent movies the friend has watched.
+                        Movie dictionaries has the following key-value pairs:
+                            title (str)
+                            genre (str)
+                            rating (float)
 
-        
+    Returns:
+        list of dict: Movies unique to the friends (movies the user has not watched). Each dictionary represents a movie. The movie dictionaries has the following key-value pairs:
+            title (str)
+            genre (str)
+            rating (float)
+    """ 
+    user_watched_movies = []
+    friends_watched_movies = []
+    yes_friend_no_user = []
+
+    for movie in user_data["watched"]:
+        user_watched_movies.append(movie)
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_watched_movies:
+                friends_watched_movies.append(movie)

```

Comment:
Unlike the preceding function, the separate friends could have the same movie in their respective lists. With this consideration in mind, you could have opted for a `set` to intrinsically ensure that no duplicate movie titles are being added. 

Also, you don't necessarily need the entire movie dictionary, just the `title`.